src/lib.rs

```diff
@@ -256,6 +256,19 @@ impl<T> Event<T> {
     ///
     /// In certain cases, this function emits a `SeqCst` fence before notifying listeners.
     ///
+    /// ## Return Value
+    ///
+    /// This function returns the number of [`EventListener`]s that were notified by this
+    /// call. However, in some cases, the notification will be delayed and this function will
+    /// return `0` instead of the number of listeners actually notified. Therefore, the return
+    /// value of this function should not be relied upon for soundness and should be used only
+    /// as a hint.
+    ///
+    /// If the `std` feature is disabled, spurious returns will happen under high contention,
+    /// such as when another thread is taking a while to `notify` the event. If the `std` 
+    /// feature is enabled, no spurious returns are possible, since the `std` implementation 
+    /// uses system locking primitives to ensure there is no unavoidable contention.
```

Comment:
Thanks. I still feel this can be clearer. People don't always read carefully. They'll only see the first paragraph and assume that this value is just in general unreliable. So how about something like this:

```suggestion
    /// This function returns the number of [`EventListener`]s that were notified by this call.
    ///
    /// #Caveats
    ///
    /// If the `std` feature is disabled, the notification will be delayed under high contention,
    /// such as when another thread is taking a while to `notify` the event,  and this function
    /// will return `0` instead of the number of listeners actually notified. Therefore, if the `std`
    /// feature is disabled the return value of this function should not be relied upon for
    /// soundness and should be used only as a hint.
    ///
    /// If the `std`  feature is enabled, no spurious returns are possible, since the `std`
    /// implementation uses system locking primitives to ensure there is no unavoidable
    /// contention.
```

English isn't my first language though so feel free to modify/improve the wording.

Reply:
Sounds good to me! I edited it a little and pushed it.